### PR TITLE
Fix issues in heterogeneous sample

### PR DESCRIPTION
**Pull request by anencore94**
1. update `ray-cluster.mini.yaml`
- There was an issue when using current `ray-cluster.mini.yaml`. So, I've changed to specify name for each port and add 10001 port to use external python client
  - https://github.com/ray-project/kuberay/issues/42#issuecomment-931015809

2. update `ray-cluster.heterogeneous.yaml`
- The original `ray-cluster.heterogeneous.yaml` doesn't work in raw. So I changed some fields to make it work.
- I'll specify each line with some screenshots.

Please take a look and give some feedback :)

### CODE REVIEW

File: ray-operator/config/samples/ray-cluster.heterogeneous.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ metadata:
     # An unique identifier for the head node and workers of this cluster.
   name: raycluster-heterogeneous
 spec:
-  rayVersion: '1.0.1'
+  rayVersion: '1.6.0'
```
Comment (anencore94) [inline ray-operator/config/samples/ray-cluster.heterogeneous.yaml:9]: I'm not sure, but the comment and image in ray-cluster.mini.yaml was rayVersion 1.6.0

File: ray-operator/config/samples/ray-cluster.heterogeneous.yaml [hunk 2]
```diff
@@ -40,22 +40,20 @@ spec:
         annotations:
           key: value
       spec:
-        initContainers: # to avoid worker crashing before head service is created
+#        initContainers: # to avoid worker crashing before head service is created
```
Comment (anencore94) [inline ray-operator/config/samples/ray-cluster.heterogeneous.yaml:43]: ray-cluster-hetero-error

When deploying this file in raw, above error was occured. So I deleted this field

File: ray-operator/config/samples/ray-cluster.heterogeneous.yaml [hunk 3]
```diff
@@ -80,8 +78,8 @@ spec:
     # if worker pods need to be removed, we decrement the replicas, and populate the podsToDelete list
     # the operator will remove pods from the list until the number of replicas is satisfied
     # when a pod is confirmed to be deleted, its name will be removed from the list below
-    scaleStrategy:
-      workersToDelete:
+#    scaleStrategy:
+#      workersToDelete:
```
Comment (anencore94) [inline ray-operator/config/samples/ray-cluster.heterogeneous.yaml:82]: ray-cluster-hetero-error

When deploying this file in raw, above error was occured. So I deleted this field, same as above

File: ray-operator/config/samples/ray-cluster.heterogeneous.yaml [hunk 4]
```diff
@@ -40,22 +40,20 @@ spec:
...
         containers:
         - name: ray-head
           image: rayproject/autoscaler
           # you can have any command and args here to run your code. 
           # the below command/args will be appended after the Ray start command and it args, and executed after Ray start.
-          command: ["python"]
+          command: ["sleep"]
           args:
-          - '/opt/code.py'
+          - 'infinity'
```
Comment (anencore94) [inline ray-operator/config/samples/ray-cluster.heterogeneous.yaml:51]: using this configmap doesn't work with rayproject/autoscaler image as above error. (maybe ray version in rayproject/autoscaler:latest image is weird)
So I deprecate to do not run `python /opt/code.py`

File: ray-operator/config/samples/ray-cluster.heterogeneous.yaml [hunk 5]
```diff
@@ -152,8 +150,8 @@ spec:
     # if worker pods need to be removed, we decrement the replicas, and populate the podsToDelete list
     # the operator will remove pods from the list until the number of replicas is satisfied
     # when a pod is confirmed to be deleted, its name will be removed from the list below
-    scaleStrategy:
-      workersToDelete:
+#    scaleStrategy:
+#      workersToDelete:
```
Comment (anencore94) [inline ray-operator/config/samples/ray-cluster.heterogeneous.yaml:154]: ray-cluster-hetero-error

When deploying this file in raw, above error was occured. So I deleted this field, same as above

File: ray-operator/config/samples/ray-cluster.heterogeneous.yaml [hunk 6]
```diff
@@ -173,7 +171,7 @@ spec:
         initContainers:
         - name: init-myservice
           image: busybox:1.28
-          command: ['sh', '-c', "until nslookup raycluster-heterogeneous-my-svc.$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace).svc.cluster.local; do echo waiting for myservice; sleep 2; done"]   
+          command: ['sh', '-c', "until nslookup raycluster-heterogeneous-head-svc.$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace).svc.cluster.local; do echo waiting for myservice; sleep 2; done"]
```
Comment (anencore94) [inline ray-operator/config/samples/ray-cluster.heterogeneous.yaml:174]: the default svc name was not `raycluster-heterogeneous-my-svc`, It was `raycluster-heterogeneous-head-svc`.
So there was an error as follows:
hetero-medium-svc-name-wrong

hetero-default-svc-name

File: ray-operator/config/samples/ray-cluster.heterogeneous.yaml (hunk 4 above)
Comment (Jeffwan) [reply ray-operator/config/samples/ray-cluster.heterogeneous.yaml:51]: The reason you meet this failure is this requires user to create configmap.yaml ealier.

@akanso WDYT? Does Ray core has example code to use so we don't need to mount our own codes?

File: ray-operator/config/samples/ray-cluster.heterogeneous.yaml (hunk 6 above)
Comment (Jeffwan) [reply ray-operator/config/samples/ray-cluster.heterogeneous.yaml:181]: This is a nice catch!  Thanks!

File: ray-operator/config/samples/ray-cluster.heterogeneous.yaml (hunk 1 above)
Comment (Jeffwan) [reply ray-operator/config/samples/ray-cluster.heterogeneous.yaml:9]: This is fine. Currently, we reserve this field and don't consider the compatibility across version.

File: ray-operator/config/samples/ray-cluster.heterogeneous.yaml (hunk 4 above)
Comment (anencore94) [reply ray-operator/config/samples/ray-cluster.heterogeneous.yaml:51]: > The reason you meet this failure is this requires user to create configmap.yaml ealier.

No, actually it isn't. I've already meet that fail case when I didn't create configmap.yaml file earlier. In that case, raycluster-heterogeneous-head pod stuck at CONTAINERCREATING with following error:
cm없을때

So, As you can see the earlier screenshot's heterogeneous-head pod status changed from RUNNING to ERROR, so I could see log above. (If pod stuck at CONTAINERCREATING, I couldn't see pod's log at all.)

There If configmap wasn't mounted, the error should be as above. And if the configmap.yaml was something wrong, the error should be something different such as 'there is no such file at /opt/code.py'. That's why I suspect the ray version in rayproject/autoscaler:latest image is weird.

I used the configmap.yaml file from [here](https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/config-map-ray-code.yaml)

File: ray-operator/config/samples/ray-cluster.heterogeneous.yaml (hunk 1 above)
Comment (anencore94) [reply ray-operator/config/samples/ray-cluster.heterogeneous.yaml:9]: I got it. Thanks :)

File: ray-operator/config/samples/ray-cluster.heterogeneous.yaml (hunk 3 above)
Comment (Jeffwan) [reply ray-operator/config/samples/ray-cluster.heterogeneous.yaml:89]: em. This is a nice catch. I think we forget to comment out these two lines.

File: ray-operator/config/samples/ray-cluster.heterogeneous.yaml (hunk 2 above)
Comment (Jeffwan) [reply ray-operator/config/samples/ray-cluster.heterogeneous.yaml:43]: nice catch

File: ray-operator/config/samples/ray-cluster.heterogeneous.yaml (hunk 4 above)
Comment (Jeffwan) [reply ray-operator/config/samples/ray-cluster.heterogeneous.yaml:51]: sound good to me.

**Comment by Jeffwan** [comment]
@anencore94 Thanks for the contribution. We are still in early stage. Feel free to raise more feature request to meet your need.

**Pull request merged**